ivy/functional/backends/numpy/elementwise.py

```diff
@@ -11,7 +12,7 @@
     _erf = None
 
 
-def add(x1: np.ndarray, x2: np.ndarray) -> np.ndarray:
+def add(x1: Union[float, np.ndarray], x2: Union[float, np.ndarray]) -> np.ndarray:
     if hasattr(x1, "dtype") and hasattr(x2, "dtype"):
         promoted_type = np.promote_types(x1.dtype, x2.dtype)
         x1, x2 = np.asarray(x1), np.asarray(x2)
```

Comment:
the line `x2 = np.asarray(x2, dtype=x1.dtype)` below will throw an error if x1 is not an array.